lib/services/role.go

```diff
@@ -1569,6 +1578,14 @@ func (set RoleSet) CheckLoginDuration(ttl time.Duration) ([]string, error) {
 	if !matchedTTL {
 		return nil, trace.AccessDenied("this user cannot request a certificate for %v", ttl)
 	}
+	if len(logins) == 0 && !set.hasPossibleLogins() {
```

Comment:
Won't this make this check always pass? I'd either look into making sure `services.RoleSet` has a role with a (dummy) login or maybe passing a flag to this function that control this behavior.

Reply:
No, the `logins` list there has been filtered down.  This check is used to distinguish between the case where the user *does* have logins, but none match, and the case where the user actually has no logins.  We're only generating the dummy login for the case that wasn't previously allowed (actually having no logins), which keeps us backwards-compatible.